python/cudf/cudf/utils/docutils.py

```diff
@@ -27,23 +27,24 @@ def docfmt(**kwargs):
 
     def outer(fn):
         buf = []
-        formatsiter = string.Formatter().parse(fn.__doc__)#
-        for literal, field, fmtspec, conv in formatsiter:
-            assert conv is None
-            assert not fmtspec
-            buf.append(literal)
-            if field is not None:
-                # get indentation
-                lines = literal.rsplit("\n", 1)
-                if _only_spaces(lines[-1]):
-                    indent = " " * len(lines[-1])
-                    valuelines = kwargs[field].splitlines(True)
-                    # first line
-                    buf.append(valuelines[0])
-                    # subsequent lines are indented
-                    buf.extend([indent + ln for ln in valuelines[1:]])
-                else:
-                    buf.append(kwargs[field])
+        if fn.__doc__ is not None:
```

Comment:
We could avoid indenting the whole of this function with any early return:

```python
if fn.__doc__  is None:
    return fn

# original code
```

Also, currently, this function will replace `None` with `""` for functions without a docstring, which I believe we don't want to do.  Using an early return will fix that too :)